src/averageAges.js

```diff
@@ -37,7 +58,15 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const womenWithChild = people.filter(person =>
+    person.sex === sexFemale && people.some(perso =>
+      perso.mother === person.name)
+  );
+  const women = people.filter(person => person.sex === sexFemale);
+
+  return withChildren
+    ? calculateAge(womenWithChild)
+    : calculateAge(women);
```

Comment:
Move the `withChildren` check inside the `filter`, so you don't have to go through the array unnecessarily.